multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakServiceClient.java

```diff
@@ -0,0 +1,234 @@
+    }
+    final HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
+    conn.setConnectTimeout(60000);
+    conn.setReadTimeout(60000);
+
+    try {
+      conn.setRequestMethod(method);
+      // drop a hint for server side that we want to receive application/json
+      conn.addRequestProperty(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON);
+      if (authToken != null) {
+        conn.setRequestProperty(HttpHeaders.AUTHORIZATION, "bearer " + authToken);
+      }
+      final int responseCode = conn.getResponseCode();
+      if ((responseCode / 100) != 2) {
+        InputStream in = conn.getErrorStream();
+        if (in == null) {
+          in = conn.getInputStream();
+        }
+        final String str;
+        try (Reader reader = new InputStreamReader(in)) {
+          str = CharStreams.toString(reader);
+        }
+        final String contentType = conn.getContentType();
+        if (contentType != null
+            && (contentType.startsWith(MediaType.APPLICATION_JSON)
+                || contentType.startsWith("application/vnd.api+json"))) {
+          final KeycloakErrorResponse serviceError =
+              DtoFactory.getInstance().createDtoFromJson(str, KeycloakErrorResponse.class);
+          if (responseCode == Response.Status.FORBIDDEN.getStatusCode()) {
+            throw new ForbiddenException(serviceError.getErrorMessage());
+          } else if (responseCode == Response.Status.NOT_FOUND.getStatusCode()) {
+            throw new NotFoundException(serviceError.getErrorMessage());
+          } else if (responseCode == Response.Status.UNAUTHORIZED.getStatusCode()) {
+            throw new UnauthorizedException(serviceError.getErrorMessage());
+          } else if (responseCode == Response.Status.CONFLICT.getStatusCode()) {
+            throw new ConflictException(serviceError.getErrorMessage());
+          } else if (responseCode == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()) {
+            throw new ServerException(serviceError.getErrorMessage());
+          } else if (responseCode == Response.Status.BAD_REQUEST.getStatusCode()) {
+            throw new BadRequestException(serviceError.getErrorMessage());
+          }
+          throw new ServerException(serviceError.getErrorMessage());
+        }
+        // Can't parse content as json or content has format other we expect for error.
+        throw new IOException(
+            String.format(
+                "Failed access: %s, method: %s, response code: %d, message: %s",
+                UriBuilder.fromUri(url).replaceQuery("token").build(), method, responseCode, str));
+      }
+      final String contentType = conn.getContentType();
+      if (responseCode != HttpURLConnection.HTTP_NO_CONTENT
+          && contentType != null
+          && !(contentType.startsWith(MediaType.APPLICATION_JSON)
+              || contentType.startsWith("application/vnd.api+json"))) {
+        throw new IOException(conn.getResponseMessage());
```

Comment:
Maybe it is better to throw `BadRequestException` with message like 'unsupported format' or smth like that, wdyt?